### PR TITLE
Complicated but...

### PR DESCRIPTION
**Pull request by ryanmitchell**
To attempt to explain.

Work out the difference between timeslot and starting time, moduled by the interval... and it gives us the correct timeslots.

### CODE REVIEW

**Comment by ryanmitchell** [comment]
@sampoyigi so pretty happy that this is correct and functional at this point. There was a random edge case where late night openings + closed the next day caused issues, but I added a work around.

**Comment by ryanmitchell** [comment]
Looks like this is (finally) ready to review and merge @sampoyigi

File: src/Location/WorkingSchedule.php [hunk 1]
```diff
@@ -317,27 +325,44 @@ public function checkStatus($dateTime = null)
...
+
+                    $timeslots[$dateString][$timeslot->getTimestamp()] = $timeslot;
+
+                }
+
+            }
         }
 
-        return collect($timeslot);
+        return collect($timeslots);
```
Comment (sampoyigi) [inline src/Location/WorkingSchedule.php:365]: Trying to wrap my head around this, can you please explain the changes you've made here, what the issue was and what was resolved.. I know there's a related issue but its a long thread.

**Comment by sampoyigi** [comment]
I understand you both have put a lot of effort getting this to work, just one more test before I made. I also refactored the code a bit, the logic remains untouched.

**Comment by BreakSecurity** [comment]
We set up a restaurant with a closing after midnight (so timeslots until 3 am). Customers do NOT have the option to pre-order their orders (as you can see from the second picture). I would therefore expect that the only timeslots shown the next day are those until 3 am.

immagine
Instead, for the day after, ALL the timeslots of that day are shown until midnight of that day (so you can book in advance for the following evening for example)

immagine
@sampoyigi

File: src/Location/WorkingSchedule.php [hunk 2]
```diff
@@ -436,7 +436,8 @@ protected function isTimeslotValid(DateTimeInterface $date, DateTimeInterface $d
         if (Carbon::now()->diffInMinutes($date) < $leadTimeMinutes)
             return FALSE;
 
-        if (Carbon::instance($dateTime)->addDays($this->days + 2)->lt($date)) // +2 as we subtracted a day and need to count the current day
+        // +2 as we subtracted a day and need to count the current day
+        if (Carbon::instance($dateTime)->addDays($this->days + 2)->lt($date))
```
Comment (BreakSecurity) [inline src/Location/WorkingSchedule.php:440]: Here the +2 I was talking about sam

**Comment by sampoyigi** [comment]
@ryanmitchell @BreakSecurity is this safe to merge now?

**Comment by BreakSecurity** [comment]
Tested and it works flawless good job!

**Pull request merged**